clc/metrics/metric_store_test.go

```diff
@@ -0,0 +1,220 @@
+				var k storageKey
+				var v int
+				check.Must(k.Unmarshal(keyb))
+				check.Must(json.Unmarshal(valb, &v))
+				entries[k] = v
+				return true, nil
+			}))
+			require.EqualValues(t, expectedEntries, entries)
+			return true
+		})
+	})
+}
+
+func TestMetricStore_Send_Today(t *testing.T) {
+	now := time.Now()
+	cid := "cid"
+	todayKey := newSimpleTestKey("map", now, cid)
+	todayValue := 5
+	WithMetricStore(func(ms *metricStore, sentQueries *[]Query) {
+		// write the entry to database
+		kb := check.MustValue(todayKey.Marshal())
+		vb := check.MustValue(json.Marshal(todayValue))
+		_ = WithStore(ms, func(s *store.Store) bool {
+			check.Must(s.SetEntry(kb, vb))
+			return true
+		})
+		// send the entries
+		check.Must(ms.Send(context.Background()))
+		// check that today's data is not sent and data exists in the database
+		_, ft := findQuery(sentQueries, todayKey.Date(), cid)
+		require.Equal(t, false, ft)
+		keysToday := WithStore(ms, func(s *store.Store) [][]byte {
+			return check.MustValue(s.GetKeysWithPrefix(datePrefix(todayKey.Date())))
+		})
+		require.Len(t, keysToday, 1)
+	})
+}
+
+func TestMetricStore_Send_Yesterday(t *testing.T) {
+	now := time.Now()
+	cid := "cid"
+	yesterdayCID := newSimpleTestKey("map", now.Add(-24*time.Hour), cid)
+	yesterday := newSimpleTestKey("map", now.Add(-24*time.Hour), "")
+	WithMetricStore(func(ms *metricStore, sentQueries *[]Query) {
+		entries := map[storageKey]int{
+			yesterdayCID: 10,
+			yesterday:    20,
+		}
+		_ = WithStore(ms, func(s *store.Store) bool {
+			for k, v := range entries {
+				kb := check.MustValue(k.Marshal())
+				vb := check.MustValue(json.Marshal(v))
+				check.Must(s.SetEntry(kb, vb))
+			}
+			return true
+		})
+		check.Must(ms.Send(context.Background()))
+		// check that yesterday's data is sent and data is deleted from the database
+		queryWithCID, ok := findQuery(sentQueries, yesterday.Date(), cid)
+		require.Equal(t, true, ok)
+		query, ok := findQuery(sentQueries, yesterday.Date(), "")
+		require.Equal(t, true, ok)
+		keysYesterday := WithStore(ms, func(s *store.Store) [][]byte {
+			return check.MustValue(s.GetKeysWithPrefix(datePrefix(yesterday.Date())))
+		})
+		require.Empty(t, keysYesterday)
+		// check that queries are formed correctly
+		require.Equal(t, 10, queryWithCID.MapRunCount)
+		require.Equal(t, 20, query.MapRunCount)
+
+	})
+}
+
+func findQuery(queries *[]Query, date, cid string) (Query, bool) {
+	for _, q := range *queries {
+		if q.Date == date && q.ClusterUUID == cid {
+			return q, true
+		}
+	}
+	return Query{}, false
+}
+
+func WithTempDir(fn func(string)) {
+	dir, err := os.MkdirTemp("", "clc-metric-store-*")
+	if err != nil {
+		panic(fmt.Errorf("creating temp dir: %w", err))
+	}
+	defer func() {
+		// errors are ignored
+		os.RemoveAll(dir)
+	}()
+	fmt.Println(dir)
+	fn(dir)
+}
+
+func WithMetricStore(fn func(ms *metricStore, queries *[]Query)) {
```

Comment:
If I don't give a pointer for the slice. When the queries is filled with values in `metricStore.Send` function it will show the old slice since its capacity is grown and points to another list.